### PR TITLE
Docs: Fix typos, grammar issues, and an invalid command

### PR DESCRIPTION
**Pull request by Legorooj**
* Removed the `the` in `the Docutils`; That's improper grammar.
* Fixed the warnings paragraph grammar errors.
* Fixed invalid command, `make clean html` - doesn't work.

### CODE REVIEW

File: doc/development/index.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ Development Guide
    coding-conventions
    testing
    commit-messages
-   documention
+   documentation
```
[the remark this answers is not in the document]
Comment (Legorooj) [reply doc/development/index.rst]: @bjones1 yes I did... Apparently I didn't commit the file. Change en route

File: doc/development/documention.rst [hunk 2]
```diff
@@ -31,12 +31,12 @@ like this::
...
-In your browser check if the markup is
+Please watch out for any warnings and errors while building the documentation.
+In your browser check if the markup is valid prior to pushing your changes and creating the pull-request please run::
 
-Prior to pushing your changes and creating the pull-request please run::
-
-  make clean html
+  make clean
+  ...
+  make html
```
Comment (htgoebel) [inline doc/development/documention.rst]: I assume this is what you refer to as the "invalid commend". I'm just curious, why this should be invalid.

Comment (Legorooj) [reply doc/development/documention.rst]: `make clean html` just cleans and ignores the second argument on windows. Not sure on Linux, but best to be safe for both.

Comment (Legorooj) [reply doc/development/documention.rst]: It's because on windows, `make.bat` is run, not `Makefile`. They're different scripts.

**Comment by htgoebel** [comment]
Thansk for this pull-request. I just merged it.

**Pull request merged**